client/src/js/components/bhProjectSelect.js

```diff
@@ -0,0 +1,47 @@
+angular.module('bhima.components')
+  .component('bhProjectSelect', {
+    templateUrl : 'modules/templates/bhProjectSelect.tmpl.html',
+    controller  : projectSelectController,
+    transclude  : true,
+    bindings    : {
+      projectId        : '<',
+      disable          : '<?',
+      onSelectCallback : '&',
+      name             : '@?',
+      required         : '<?',      
+    },
+  });
+
+projectSelectController.$inject = [
+  'ProjectService', 'NotifyService'
+];
+
+/**
+ * Project Select component
+ *
+ */
+function projectSelectController(Projects, Notify) {
+  var $ctrl = this;
+
+  $ctrl.$onInit = function onInit() {    
+    // fired when a project has been selected
+    $ctrl.onSelectCallback = $ctrl.onSelectCallback || angular.noop;
```

Comment:
You don't actually need this - angular will automatically assign a `noop()` to it since it so not required (`<?`).